x-pack/filebeat/module/okta/system/config/pipeline.js

```diff
@@ -0,0 +1,166 @@
+            { from: "json.client.zone", to: "okta.client.zone" },
+            { from: "json.outcome.reason", to: "okta.outcome.reason" },
+            { from: "json.outcome.result", to: "okta.outcome.result" },
+            { from: "json.target", to: "okta.target" },
+            { from: "json.transaction.id", to: "okta.transaction.id" },
+            { from: "json.transaction.type", to: "okta.transaction.type" },
+            { from: "json.debugContext.debugData.deviceFingerprint", to: "okta.debug_context.debug_data.device_fingerprint" },
+            { from: "json.debugContext.debugData.requestId", to: "okta.debug_context.debug_data.request_id" },
+            { from: "json.debugContext.debugData.requestUri", to: "okta.debug_context.debug_data.request_uri" },
+            { from: "json.debugContext.debugData.threatSuspected", to: "okta.debug_context.debug_data.threat_suspected" },
+            { from: "json.debugContext.debugData.url", to: "okta.debug_context.debug_data.url" },
+            { from: "json.authenticationContext.authenticationProvider", to: "okta.authentication_context.authentication_provider" },
+            { from: "json.authenticationContext.authenticationStep", to: "okta.authentication_context.authentication_step" },
+            { from: "json.authenticationContext.credentialProvider", to: "okta.authentication_context.credential_provider" },
+            { from: "json.authenticationContext.credentialType", to: "okta.authentication_context.credential_type" },
+            { from: "json.authenticationContext.externalSessionId", to: "okta.authentication_context.external_session_id" },
+            { from: "json.authenticationContext.interface", to: "okta.authentication_context.authentication_provider" },
+            { from: "json.authenticationContext.issuer", to: "okta.authentication_context.issuer" },
+            { from: "json.securityContext.asNumber", to: "okta.security_context.as.number" },
+            { from: "json.securityContext.asOrg", to: "okta.security_context.as.organization.name" },
+            { from: "json.securityContext.domain", to: "okta.security_context.domain" },
+            { from: "json.securityContext.isProxy", to: "okta.security_context.is_proxy" },
+            { from: "json.securityContext.isp", to: "okta.security_context.isp" },
+        ],
+        mode: "rename",
+        ignore_missing: true,
+        fail_on_error: false,
+    });
+
+    var copyFields = new processor.Convert({
+        fields: [
+            { from: "okta.client.user_agent.raw_user_agent", to: "user_agent.original" },
+            { from: "okta.client.ip", to: "client.ip" },
+            { from: "okta.client.ip", to: "source.ip" },
+            { from: "okta.event_type", to: "event.action" },
+            { from: "okta.outcome.result", to: "event.outcome" },
+            { from: "okta.security_context.as", to: "client.as" },
```

Comment:
In the golden files it looks like this brought over some null values.  It would be nice to avoid that.  I wonder if we should walk the json object and delete any keys that are set to null or empty string, before we decode_json_fields.

Reply:
Added a function to drop all null fields in the end.